src/mission_companion.cpp

```diff
@@ -162,42 +205,40 @@ void talk_function::bionic_remove(npc &p)
 
 void talk_function::companion_mission(npc &p)
 {
- std::string id = "NONE";
- std::string title = _("Outpost Missions");
- unsigned int a = -1;
- // Name checks determining role? Horrible!
- if (p.name.find("Scavenger Boss") != a){
-    id = "SCAVENGER";
-    title = _("Junk Shop Missions");
- }
- if (p.name.find("Crop Overseer") != a){
-    id = "COMMUNE CROPS";
-    title = _("Agricultural Missions");
- }
- if (p.name.find("Foreman") != a){
-    id = "FOREMAN";
-    title = _("Construction Missions");
- }
- if (p.name.find(", Merchant") != a){
-    id = "REFUGEE MERCHANT";
-    title = _("Free Merchant Missions");
- }
- talk_function::outpost_missions( p, id, title );
+    std::string id = p.companion_mission_role_id;
+    std::string title = _("Outpost Missions");
+    if( id == "FACTION_CAMP" ){
+        title = _("Base Missions");
+    } else if( id == "SCAVENGER" ){
+        title = _("Junk Shop Missions");
+    } else if( id == "COMMUNE CROPS" ){
+        title = _("Agricultural Missions");
+    } else if( id == "FOREMAN" ){
+        title = _("Construction Missions");
+    } else if( id == "REFUGEE MERCHANT" ){
+        title = _("Free Merchant Missions");
+    }
```

Comment:
Now THIS is a place where you actually want to use a `std::map`.
```
std::map<std::string, std::string> titles = {{
    { "FACTION_CAMP", "Base Missions" }
}};
const auto iter = std::find( titles.begin(), titles.end() id );
if( iter != titles.end() ) {
    title = iter->second;
}
```

Preferably wrapped in a function, for easier reuse.